### PR TITLE
#45 type safe localstorage

### PR DESCRIPTION
**Pull request by joshxg**
#45 

- getItem / setItem should be typed dependent on the key passed and the key in `keys`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ module.exports = {
   settings: { react: { version: 'detect' } },
   plugins: ['react', , '@typescript-eslint'],
   rules: {
+    camelcase: ['error', { allow: ['_ONLY_FOR_TESTING'] }],
```
Comment (joshxg) [inline .eslintrc.js:31]: 🤷 is this a good idea? functions only to be used for testing having this suffix..?

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "compile": "lerna run --scope @openmsupply-client/* --parallel tsc",
     "prepare": "husky install",
     "pre-commit-lint": "yarn compile && yarn lint-staged",
-    "test": "jest --config ./jest.config.js",
+    "test": "jest --config ./jest.config.js --maxWorkers=50%",
```
Comment (joshxg) [inline package.json:11]: why are the jest tests so slow!

**Review by joshxg — COMMENTED** [review]
Hm. I like this, but I can see a few things that are debatable!

**Review by mark-prins — APPROVED** [review]
very nifty. certainly like the addition of type safety on the local storage access, I think it will be useful to have local storage so worth getting usage tidy.

**Pull request merged**